### PR TITLE
Use the appropriate config dir for the registry

### PR DESCRIPTION
**Pull request by samueldr**
It was reported on IRC that Nix would create a `.config` dir even though  `XDG_CONFIG_HOME` was set.

This is the only occurence of `.config` I could grep for that wasn't in `getConfigDir()`.

The test is a bit... ugly... And might be missing some other actions that could create the config dir...

But otherwise, it **does** test this particular case. Reverting the fix (tip of this branch) makes it fail as expected.

I guess we want to test other commands that can write to the config directory, are there others?

We probably also want to test the equivalent for the different caches.

### CODE REVIEW

**Review by Ericson2314 — APPROVED** [review]

File: tests/config.sh [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+[ ! -e "/tmp/confighome/nix" ]
+nix registry remove userhome-with-xdg
+# Verifies the confighome path has been created
+[ -e "/tmp/confighome/nix" ]
+# Assert the .config folder hasn't been created.
+[ ! -e "$HOME/.config" ]
+
 # Test that files are loaded from XDG by default
-export XDG_CONFIG_HOME=/tmp/home
+export XDG_CONFIG_HOME=/tmp/confighome
```
Comment (samueldr) [inline tests/config.sh:30]: Yes, this export is now redundant considering the previous export.

The following tests, though, are not redundant. None are.

**Comment by edolstra** [comment]
We shouldn't use fixed paths like `/tmp/home` in the tests since that's racy and breaks in multi-user environments. I see that recently got introduced in 895516cadf0739dd8359a38d503e0aaf5068dc4b but it's a mistake.

**Comment by samueldr** [comment]
Totally agree that it may be racy. I even changed it to reduce raciness assuming (verified through grepping) that the path is unique between all tests.

Though I see it's not foolproof. What strategy should be used to isolate each test files?

**Comment by samueldr** [comment]
Are we blocking this on the fixed paths?

If so, I'll need a suggestion to make this go forward. I'm not sure what approach could be used in these tests.

If we're not blocking on the fixed paths, anything else needs to be changed?

**Comment by regnat** [comment]
> What strategy should be used to isolate each test files?

I guess using `$TEST_ROOT` is good-enough (it's doesn't really bring much more guaranty if the test is run twice concurrently, but it's what everything else uses, so at least there's only one place to change if must be)

**Comment by samueldr** [comment]
Applied the fix to use `$TEST_ROOT`.

 - [x] Tested the new test still fails without the fix (the last commit)
 - [x] Tested the tests work with the fix

**Comment by edolstra** [comment]
Thanks @samueldr!

**Pull request merged**